src/hooks/web3.ts

```diff
@@ -57,17 +69,25 @@ export function useInactiveListener(suppress = false) {
     if (ethereum && ethereum.on && !active && !error && !suppress) {
       const handleChainChanged = () => {
         // eat errors
-        activate(injected, undefined, true).catch((error) => {
-          console.error('Failed to activate after chain changed', error)
-        })
+        activate(injected, undefined, true)
+          .catch((error) => {
+            console.error('Failed to activate after chain changed', error)
+          })
+          .finally(() =>
+            generateAffiliateLink((link: string) => dispatch(updateAffiliateLink({ affiliateLink: link })))
+          )
       }
 
       const handleAccountsChanged = (accounts: string[]) => {
         if (accounts.length > 0) {
           // eat errors
-          activate(injected, undefined, true).catch((error) => {
-            console.error('Failed to activate after accounts changed', error)
-          })
+          activate(injected, undefined, true)
+            .catch((error) => {
+              console.error('Failed to activate after accounts changed', error)
+            })
+            .finally(() =>
+              generateAffiliateLink((link: string) => dispatch(updateAffiliateLink({ affiliateLink: link })))
+            )
```

Comment:
Is this necessary? Remember that this file is not under the `custom` folder.

Reply:
@ramirotw do you want to move this file to custom folder?
It is used in too many files already.